nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -362,4 +372,73 @@ final class BinaryDeserializer(buffer: ByteBuffer, bufferName: String) {
 
     readPosition()
   }
+
+  // Copy pasted from Scala 3.1.0 compiler
+  // Used to encode names from NIR versions <= 5.8 produced with decoded names
+  private object NameTransformer {
```

Comment:
There is already `scala.reflect.NameTransformer` in the standard library, which provides this functionality.